lib/eblox/data/provider.ex

```diff
@@ -53,33 +58,34 @@ defmodule Eblox.Data.Provider do
   def perform(:died, _id, payload), do: {:transition, :*, payload}
   def perform(_state, _id, payload), do: {:transition, :scan, payload}
 
-  @spec handle_changes(t()) :: [:ok]
-  def handle_changes(%Provider{created: _, deleted: _, changed: _} = changes) do
+  @spec handle_changes(module(), t()) :: [:ok]
+  def handle_changes(impl, %Provider{created: _, deleted: _, changed: _} = changes) do
     changes
     |> Flow.from_enumerable()
     |> Flow.flat_map(fn {action, list} -> Enum.map(list, &{action, &1}) end)
     |> Flow.partition()
-    |> Flow.reduce(fn -> [] end, fn {action, elem}, acc -> [action(action, elem) | acc] end)
+    |> Flow.reduce(fn -> [] end, fn {action, elem}, acc -> [action(impl, action, elem) | acc] end)
     |> Enum.to_list()
```

Comment:
This is ridiculous, btw. l know this code is not a part of the PR, but please amend it to [`Stream.run/1`](https://hexdocs.pm/elixir/Stream.html#run/1) which is to be used for side effects. (Also, the return type of the function would be `:ok`.)
```suggestion
    |> Stream.run()
```